.github/scripts/prereq.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+mkdir -p "$HOME"/.local/bin
+
+export PATH="$HOME/.local/bin:$PATH"
+
+export BOOTSTRAP_HASKELL_NONINTERACTIVE=1
+export BOOTSTRAP_HASKELL_GHC_VERSION="${GHC_VER:-recommended}"
+export BOOTSTRAP_HASKELL_CABAL_VERSION="${CABAL_VER:-recommended}"
+export BOOTSTRAP_HASKELL_ADJUST_CABAL_CONFIG=yes
+export BOOTSTRAP_HASKELL_INSTALL_NO_STACK=yes
+
+if [ "${RUNNER_OS}" = "Windows" ] ; then
+	# on windows use pwd to get unix style path
+	CI_PROJECT_DIR="$(pwd)"
+	export CI_PROJECT_DIR
+    export GHCUP_INSTALL_BASE_PREFIX="/c"
+    export GHCUP_BIN="$GHCUP_INSTALL_BASE_PREFIX/ghcup/bin"
+    export PATH="$GHCUP_BIN:$PATH"
+	export CABAL_DIR="C:\\Users\\runneradmin\\AppData\\Roaming\\cabal"
+else
+	export CI_PROJECT_DIR="${GITHUB_WORKSPACE}"
+    export GHCUP_INSTALL_BASE_PREFIX="$CI_PROJECT_DIR"
+    export GHCUP_BIN="$GHCUP_INSTALL_BASE_PREFIX/.ghcup/bin"
+    export PATH="$GHCUP_BIN:$PATH"
+    export CABAL_DIR="$CI_PROJECT_DIR/cabal"
+    export CABAL_CACHE="$CI_PROJECT_DIR/cabal-cache"
+fi
+
+export DEBIAN_FRONTEND=noninteractive
+export TZ=Asia/Singapore
+
+if [ "${RUNNER_OS}" = "macOS" ] ; then
+	if ! command -v brew ; then
```

Comment:
I would prefer that we use Nix (like GHC CI) for building on darwin as it more closely matches the environment that GHC is built in, and it also seems much less likely to break than brew. There is already nix configuration in `.gitlab/darwin/nix` that you can use in exactly the same way we did in gitlab.

Reply:
I will not be working on nix and I disagree that it's necessary. 

We have been using brew for ghcup and cabal and stack and HLS binaries in the past. It works fine.

Reply:
It didn't work fine for the last release when brew upgraded llvm to version 14, which ended up with us being unable to release 1.8.0.1.

You don't need to work on Nix, the existing Nix setup bits just needs to be copied over from the gitlab CI script.

As release manager, I do not want to manage a seperate brew configuration for HLS when GHC's nix configuration is tested every day in GHC CI and I can be assured that it works.

Reply:
> It didn't work fine for the last release when brew upgraded llvm to version 14, which ended up with us being unable to release 1.8.0.1.

You can install a specific llvm version via brew, which is what we're doing here.

```
    if [ "${ARCH}" = "ARM64" ] ; then
		brew install llvm@13
		export PATH="$HOME/.brew/opt/llvm@13/bin:$PATH"
		export CC="$HOME/.brew/opt/llvm@13/bin/clang"
		export CXX="$HOME/.brew/opt/llvm@13/bin/clang++"
		export LD=ld
		export AR="$HOME/.brew/opt/llvm@13/bin/llvm-ar"
		export RANLIB="$HOME/.brew/opt/llvm@13/bin/llvm-ranlib"
	fi
```

Any issues with that? Why do I need nix for that?

Reply:
Because we have to figure out the right incantations for whichever version of a tool breaks.